httpe/method.go

```diff
@@ -0,0 +1,78 @@
+package httpe
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var CheckGet = NewMethodCheck(http.MethodGet)
+var CheckPost = NewMethodCheck(http.MethodPost)
+var CheckPut = NewMethodCheck(http.MethodPut)
+var CheckPatch = NewMethodCheck(http.MethodPatch)
+var CheckDelete = NewMethodCheck(http.MethodDelete)
+
+func NewMethodCheck(method string) HandlerFuncE {
+	return HandlerFuncE(func(_ http.ResponseWriter, r *http.Request) error {
+		if r.Method != method {
+			return ErrMethodNotAllowed
+		}
+		return nil
+	})
+}
+
+// string: http.Method
+type MethodMux map[string]HandlerE
+
+func (m MethodMux) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
+	h := m[r.Method]
+	if h == nil {
+		return ErrMethodNotAllowed
+	}
+	return h.ServeHTTP(w, r)
+}
+
+func NewMethodMux() MethodMux {
+	return MethodMux{}
+}
+
+func (m MethodMux) Get(h HandlerE) MethodMux {
+	m[http.MethodGet] = h
+	return m
+}
+
+func (m MethodMux) Post(h HandlerE) MethodMux {
+	m[http.MethodPost] = h
+	return m
+}
+
+func (m MethodMux) Put(h HandlerE) MethodMux {
+	m[http.MethodPut] = h
+	return m
+}
+
+func (m MethodMux) Delete(h HandlerE) MethodMux {
+	m[http.MethodDelete] = h
+	return m
+}
+
+// WriteJSON marshals struct and writes it to ResposneWriter.
```

Comment:
feels like this and `ReadJSON` belongs in `json.go`?

Reply:
yes - that's what I wanted to do. Thanks for finishing my half-finished code ;)